src/Features/CSharp/Portable/UseExpressionBody/Helpers/AbstractUseExpressionBodyHelper.cs

```diff
@@ -1,90 +1,105 @@
+
+        protected AbstractUseExpressionBodyHelper(
+            LocalizableString useExpressionBodyTitle,
+            LocalizableString useBlockBodyTitle,
+            Option<CodeStyleOption<ExpressionBodyPreference>> option)
         {
-            FixableDiagnosticIds = ImmutableArray.Create(diagnosticId);
-            _option = option;
-            _useExpressionBodyTitle = useExpressionBodyTitle;
-            _useBlockBodyTitle = useBlockBodyTitle;
+            Option = option;
+            UseExpressionBodyTitle = useExpressionBodyTitle;
+            UseBlockBodyTitle = useBlockBodyTitle;
         }
 
-        protected override bool IncludeDiagnosticDuringFixAll(Diagnostic diagnostic)
-            => diagnostic.Severity != DiagnosticSeverity.Hidden;
+        public abstract BlockSyntax GetBody(TDeclaration declaration);
+        public abstract ArrowExpressionClauseSyntax GetExpressionBody(TDeclaration declaration);
 
-        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        protected static BlockSyntax GetBodyFromSingleGetAccessor(AccessorListSyntax accessorList)
         {
-            var diagnostic = context.Diagnostics.First();
-            var documentOptionSet = await context.Document.GetOptionsAsync(context.CancellationToken).ConfigureAwait(false);
-
-            var priority = diagnostic.Severity == DiagnosticSeverity.Hidden
-                ? CodeActionPriority.Low
-                : CodeActionPriority.Medium;
+            if (accessorList != null &&
+                accessorList.Accessors.Count == 1 &&
+                accessorList.Accessors[0].AttributeLists.Count == 0 && 
+                accessorList.Accessors[0].IsKind(SyntaxKind.GetAccessorDeclaration))
+            {
+                return accessorList.Accessors[0].Body;
+            }
 
-            context.RegisterCodeFix(
-                new MyCodeAction(diagnostic.GetMessage(), priority, c => FixAsync(context.Document, diagnostic, c)),
-                diagnostic);
+            return null;
         }
 
-        protected override async Task FixAllAsync(
-            Document document, ImmutableArray<Diagnostic> diagnostics,
-            SyntaxEditor editor, CancellationToken cancellationToken)
+        public virtual bool CanOfferUseExpressionBody(
+            OptionSet optionSet, TDeclaration declaration, bool forAnalyzer)
         {
-            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var preference = optionSet.GetOption(this.Option).Value;
+            var userPrefersExpressionBodies = preference != ExpressionBodyPreference.Never;
 
-            foreach (var diagnostic in diagnostics)
+            // If the user likes expression bodies, then we offer expression bodies from the diagnostic analyzer.
+            // If the user does not like expression bodies then we offer expression bodies from the refactoring provider.
```

Comment:
I sort of dislike disabling the refactoring here. Customers already have complaints about not knowing when a refactoring is available.

I'd be inclined to just leave it as is and wait for customer feedback.